readme/examples.md

```diff
@@ -0,0 +1,154 @@
+- `if_tap(k(a), k(b))` writes a if the key is tapped, otherwise b
+- `if_tap(k(a), k(b), 1000)` writes a if the key is released within a second, otherwise b
+- `if_single(k(a), k(b))` writes b if another key is pressed, or a if the key is released
+  and no other key was pressed in the meantime.
+
+## Scroll and Click on a Keyboard
+
+Seldom used PrintScreen, ScrollLock and Pause keys on keyboards with TKL (ten key
+less) layout are easily accessible by the right hand thanks to the missing
+numeric block, so they can be mapped to mouse scroll and click events:
+
+- Print: wheel(up, 1)
+- Pause: wheel(down, 1)
+- Scroll Lock: BTN_LEFT
+- Menu: BTN_RIGHT
+- F12: KEY_LEFTCTRL+w
+
+In contrast to libinput's `ScrollMethod` `button` which requires the scroll
+button to belong to the same (mouse) device, clicking and scrolling events mapped
+to a keyboard key can fully cooperate with events from a real mouse, e.g.
+drag'n'drop by holding a (mapped) keyboard key and moving the cursor by mouse.
+
+Mapping the scrolling to a keyboard key is also useful for trackballs without
+a scroll ring.
+
+In contrast to a real scroll wheel, holding a key which has mouse wheel event
+mapped produces linear auto-repeat, without any acceleration. Using a PageDown
+key for fast scrolling requires only a small adjustment of the right hand
+position.
+
+## Scroll on a 3-Button Mouse
+
+Cheap 3-button mouse without a scroll wheel can scroll using the middle button:
+
+- Button MIDDLE: wheel(down, 1)
+
+## Click on Lower Buttons of Trackball
+
+Trackball with 4 buttons (e.g. Kensington Wireless Expert Mouse) with lower 2
+buttons by default assigned to middle and side button can be remapped to provide
+left and right click on both the upper and lower pairs of buttons to avoid
+readjusting a hand after moving the cursor down:
+
+- Button MIDDLE: BTN_LEFT
+- Button SIDE: BTN_RIGHT
+
+## Scroll on Foot Pedals
+
+While Kinesis Savant Elite 2 foot pedals can be programmed to emit key press or
+mouse click events, they cannot emit scroll events themselves. Using the pedals
+for scrolling while standing at a standing desk is possible thanks to remapping:
+
+- Button LEFT: wheel(up, 1)
+- Button RIGHT: wheel(down, 1)
+
+## Gamepads
+
+Joystick movements will be translated to mouse movements, while the second
+joystick acts as a mouse wheel. You can swap this in the user interface.
+All buttons, triggers and D-Pads can be mapped to keycodes and macros.
+
+The D-Pad can be mapped to W, A, S, D for example, to run around in games,
+while the joystick turns the view (depending on the game).
+
+Tested with the XBOX 360 Gamepad. On Ubuntu, gamepads worked better in
+Wayland than with X11.
+
+## Combinations spanning multiple devices
+
+For regular combinations on only single devices it is not required to
+configure macros. See [readme/usage.md](usage.md#combinations).
+
+**Keyboard** `space` `set(foo, bar).h(space).set(foo, 0)`
+
+**Mouse** `middle` `ifeq(foo, bar, h(a), h(BTN_MIDDLE))`
```

Comment:
```suggestion
**Keyboard** `space` `set(foo, 1).h(space).set(foo, 0)`

**Mouse** `middle` `if_eq($foo, 1, h(a), h(BTN_MIDDLE))`
```

if_eq supports the new `$variable` syntax, ifeq should not be mentioned in the docs anymore. I couldn't migrate ifeq to support that feature without breaking the way it previously worked.